atc/api/handler.go

```diff
@@ -125,18 +125,22 @@ func NewHandler(
 
 		atc.GetCheck: http.HandlerFunc(checkServer.GetCheck),
 
-		atc.ListAllJobs:    http.HandlerFunc(jobServer.ListAllJobs),
-		atc.ListJobs:       pipelineHandlerFactory.HandlerFor(jobServer.ListJobs),
-		atc.GetJob:         pipelineHandlerFactory.HandlerFor(jobServer.GetJob),
-		atc.ListJobBuilds:  pipelineHandlerFactory.HandlerFor(jobServer.ListJobBuilds),
-		atc.ListJobInputs:  pipelineHandlerFactory.HandlerFor(jobServer.ListJobInputs),
-		atc.GetJobBuild:    pipelineHandlerFactory.HandlerFor(jobServer.GetJobBuild),
-		atc.CreateJobBuild: pipelineHandlerFactory.HandlerFor(jobServer.CreateJobBuild),
-		atc.RerunJobBuild:  pipelineHandlerFactory.HandlerFor(jobServer.RerunJobBuild),
-		atc.PauseJob:       pipelineHandlerFactory.HandlerFor(jobServer.PauseJob),
-		atc.UnpauseJob:     pipelineHandlerFactory.HandlerFor(jobServer.UnpauseJob),
-		atc.ScheduleJob:    pipelineHandlerFactory.HandlerFor(jobServer.ScheduleJob),
-		atc.JobBadge:       pipelineHandlerFactory.HandlerFor(jobServer.JobBadge),
+		atc.ListAllJobs:   http.HandlerFunc(jobServer.ListAllJobs),
+		atc.ListJobs:      pipelineHandlerFactory.HandlerFor(jobServer.ListJobs),
+		atc.GetJob:        pipelineHandlerFactory.HandlerFor(jobServer.GetJob),
+		atc.ListJobBuilds: pipelineHandlerFactory.HandlerFor(jobServer.ListJobBuilds),
+		atc.ListJobInputs: pipelineHandlerFactory.HandlerFor(jobServer.ListJobInputs),
+		atc.GetJobBuild:   pipelineHandlerFactory.HandlerFor(jobServer.GetJobBuild),
+		atc.CreateJobBuild: pipelineHandlerFactory.HandlerFor(
+			pipelineHandlerFactory.RejectArchived(jobServer.CreateJobBuild),
+		),
+		atc.RerunJobBuild: pipelineHandlerFactory.HandlerFor(jobServer.RerunJobBuild),
+		atc.PauseJob:      pipelineHandlerFactory.HandlerFor(jobServer.PauseJob),
```

Comment:
Not sure what you think about pausing a job? It won't really have any impact if they pause an archived job but I'm curious if we want to block all write operation APIs?

Reply:
doesn't hurt to be consistent!